File: core/src/main/java/pcb/uwu/units/composite/electromagnetism/MagneticFieldUnit.java
package pcb.uwu.units.composite.electromagnetism;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.composite.mechanics.ForceUnit;
import pcb.uwu.units.composite.termodynamics.EnergyUnit;
import pcb.uwu.units.quantity.ElectricCurrentUnit;
import pcb.uwu.units.quantity.LengthUnit;
import pcb.uwu.units.quantity.MassUnit;
import pcb.uwu.units.quantity.TimeUnit;

public class MagneticFieldUnit extends CompositeUnit {

	// region base constructor

	public MagneticFieldUnit(MassUnit massUnit, ElectricCurrentUnit electricCurrentUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.minor(electricCurrentUnit)
				.minor(timeUnit, 2));
	}

	// endregion

	// region composite constructors

	public MagneticFieldUnit(ElectricPotentialUnit electricPotentialUnit, TimeUnit timeUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(electricPotentialUnit.getUnitCounter())
				.major(timeUnit)
				.minor(lengthUnit, 2));
	}

	public MagneticFieldUnit(ForceUnit forceUnit, ElectricCurrentUnit electricCurrentUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(forceUnit.getUnitCounter())
				.minor(electricCurrentUnit)
				.minor(lengthUnit));
	}

	public MagneticFieldUnit(EnergyUnit energyUnit, ElectricCurrentUnit electricCurrentUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(energyUnit.getUnitCounter())
				.minor(electricCurrentUnit)
				.minor(lengthUnit, 2));
	}

	public MagneticFieldUnit(ElectricInductanceUnit electricInductanceUnit, ElectricCurrentUnit electricCurrentUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(electricInductanceUnit.getUnitCounter())
				.major(electricCurrentUnit)
				.minor(lengthUnit, 2));
	}

	public MagneticFieldUnit(MagneticFluxUnit magneticFluxUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(magneticFluxUnit.getUnitCounter())
				.minor(lengthUnit, 2));
	}

	public MagneticFieldUnit(MassUnit massUnit, ElectricChargeUnit electricChargeUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.minor(electricChargeUnit.getUnitCounter())
				.minor(timeUnit));
	}

	public MagneticFieldUnit(ForceUnit forceUnit, TimeUnit timeUnit, ElectricChargeUnit electricChargeUnit, LengthUnit lengthUnit) {
		super(new UnitCounter()
				.major(forceUnit.getUnitCounter())
				.major(timeUnit)
				.minor(electricChargeUnit.getUnitCounter())
				.minor(lengthUnit));
	}

	// endregion


}

File: core/src/main/java/pcb/uwu/unit/derived/electromagnetism/MagneticFieldUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.ElectricCurrentUnit
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit
import pcb.uwu.unit.derived.fundamental.AreaUnit
import pcb.uwu.unit.derived.mechanics.ForceUnit
import pcb.uwu.unit.derived.termodynamics.EnergyUnit

open class MagneticFieldUnit : CompositeUnit
{
    constructor(massUnit: MassUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .minor(electricCurrentUnit)
                        .minor(timeUnit)
                        .minor(timeUnit))

    constructor(electricPotentialUnit: ElectricPotentialUnit,
                timeUnit: TimeUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(electricPotentialUnit)
                        .major(timeUnit)
                        .minor(lengthUnit)
                        .minor(lengthUnit))

    constructor(forceUnit: ForceUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(forceUnit)
                        .minor(electricCurrentUnit)
                        .minor(lengthUnit))

    constructor(energyUnit: EnergyUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(energyUnit)
                        .minor(electricCurrentUnit)
                        .minor(lengthUnit)
                        .minor(lengthUnit))

    constructor(electricInductanceUnit: ElectricInductanceUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(electricInductanceUnit)
                        .major(electricCurrentUnit)
                        .minor(lengthUnit)
                        .minor(lengthUnit))

    constructor(magneticFluxUnit: MagneticFluxUnit,
                areaUnit: AreaUnit)
            : super(UnitCounter()
                        .major(magneticFluxUnit)
                        .minor(areaUnit))

    constructor(massUnit: MassUnit,
                electricChargeUnit: ElectricChargeUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .minor(electricChargeUnit)
                        .minor(timeUnit))

    constructor(forceUnit: ForceUnit,
                timeUnit: TimeUnit,
                electricChargeUnit: ElectricChargeUnit,
                lengthUnit: LengthUnit)
            : super(UnitCounter()
                        .major(forceUnit)
                        .major(timeUnit)
                        .minor(electricChargeUnit)
                        .minor(lengthUnit))
}